### PR TITLE
Fix powershell RM cmdlet generation and add some unittests

### PR DESCRIPTION
**Pull request by bala16**
Use New-AzureRmResourceGroup instead of New-AzureRmResource when creating new resource groups. Use -IsCollection flag only if resource name is empty. Fix duplicate -ResourceName parameter for RM cmdlets operating on named resources by using the format name_of_res_to_modify/name_for_new_resource_being_added.

### CODE REVIEW

File: ng/PowerShellScriptGenerator.ts [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+        SUBSCRIPTIONS_VALUE,
+        RESOURCEGROUPS_KEY,
+        RESOURCEGROUPS_VALUE, // Urls with length <= URLPARTS_RESOURCEGROUPS_VALUE will only include resourceId
+        PROVIDER_KEY,
+        PROVIDER_VALUE, // start of resourcetype followed by 9, 11, 13, ... ( 8,9 will always exists, 11,13.. is optional)
+        RESOURCETYPE1_NAME,
+        RESOURCETYPE1_VALUE // start of resourcename followed by 12, 14,...(12, 14,... is optional)
+    }
+
+    export class RMUrlParser {
```
Comment (ahmelsayed) [inline ng/PowerShellScriptGenerator.ts]: I'd call it `ARMUrlParser` as we always refer to it as ARM rather than RM.

**Comment by ahmelsayed** [comment]
wow, changes are great :) it makes the powershell code much cleaner. it would be great if the core logic for building the tree and managing the projection tree was also cleaned up like this :)

one note, it's hard to review manage.ts since there is quite a large number of whitespace\formatting changes. Can you either point to the changes there, or split the formatting and code changes in 2 different commits?

File: ng/PowerShellScriptGenerator.ts (hunk 1 above)
Comment (bala16) [inline ng/PowerShellScriptGenerator.ts]: sure. renaming this now..

File: ng/PowerShellScriptGenerator.ts [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+        BLANK,
+        DOMAINNAME,
+        SUBSCRIPTIONS_KEY,
+        SUBSCRIPTIONS_VALUE,
+        RESOURCEGROUPS_KEY,
+        RESOURCEGROUPS_VALUE, // Urls with length <= URLPARTS_RESOURCEGROUPS_VALUE will only include resourceId
+        PROVIDER_KEY,
+        PROVIDER_VALUE, // start of resourcetype followed by 9, 11, 13, ... ( 8,9 will always exists, 11,13.. is optional)
+        RESOURCETYPE1_NAME,
+        RESOURCETYPE1_VALUE // start of resourcename followed by 12, 14,...(12, 14,... is optional)
```
Comment (ahmelsayed) [inline ng/PowerShellScriptGenerator.ts]: btw, any particular reason for using all caps for enum values? I'd use regular CamelCasing for the keys if there is no explicit need for all caps.

**Comment by bala16** [comment]
thanks :) I plan on refactoring as much as i can when working on specific parts of the code. 
i agree the formatting changes makes the diff confusing. will undo all the formatting changes and push as a separate commit.

File: ng/PowerShellScriptGenerator.ts (hunk 2 above)
Comment (bala16) [inline ng/PowerShellScriptGenerator.ts]: i originally had them as individual constants and i usually have constants all caps for easy identification and just directly copied them into an enum. looks like the typescript coding guidelines also call for PascalCase like you mentioned. https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines

File: ng/manage.ts [hunk 3]
```diff
@@ -1,1639 +1,1639 @@
...
-
-            $scope.enterDataTab = () => {
-                [responseEditor, requestEditor].map((e) => {
-                    if (e) {
-                        e.resize();
+                delete $scope.createError;
+                var userObject = JSON.parse(createEditor.getValue());
+                cleanObject(userObject);
+                $scope.invoking = true;
+
```
Comment (bala16) [inline ng/manage.ts]: changed the func name and impl. this is the only change in this file. (other than removing some now unused functions and moving GETPSObjectFromJSON() to PowerShellScriptGenerator.ts file)

**Comment by ahmelsayed** [comment]
:shipit:

**Pull request merged**